include/verilated.cpp

```diff
@@ -1232,15 +1243,37 @@ IData VL_FOPEN_WI(int fnwords, WDataInP filenamep, IData mode) VL_MT_SAFE {
     return VL_FOPEN_S(filenamez, modez);
 }
 IData VL_FOPEN_S(const char* filenamep, const char* modep) VL_MT_SAFE {
-    return VerilatedImp::fdNew(fopen(filenamep, modep));
+    return VerilatedImp::fdNew(filenamep, modep);
+}
+IData VL_FOPEN_MCD_S(const char* filenamep) VL_MT_SAFE {
+    return VerilatedImp::fdNewMcd(filenamep);
+}
+IData VL_FOPEN_MCD_N(const std::string& filename) VL_MT_SAFE {
+    return VL_FOPEN_MCD_S(filename.c_str());
+}
+IData VL_FOPEN_MCD_W(int fnwords, WDataInP filenamep) VL_MT_SAFE {
+    char filenamez[VL_TO_STRING_MAX_WORDS * VL_EDATASIZE + 1];
+    _VL_VINT_TO_STRING(fnwords * VL_EDATASIZE, filenamez, filenamep);
+    return VL_FOPEN_MCD_S(filenamez);
+}
+IData VL_FOPEN_MCD_Q(QData filename) VL_MT_SAFE {
+    WData fnw[VL_WQ_WORDS_E];
+    VL_SET_WQ(fnw, filename);
+    return VL_FOPEN_MCD_W(VL_WQ_WORDS_E, fnw);
```

Comment:
I think you need only VL_FOPEN_MCD_N that is not the _Q/_W flavors.  Have V3EmitC when it puts out the filename call emitCvtPackStr. That will put out appropriate code to make sure a string is passed down.  V3EmitCInlines.cpp then needs to call needHeavy(true) when either AstFOpen is seen to make sure verilated_heavy.his properly included.

Please likewise cleanup old fopen.

(The old method predates having any std::strings being emitted.)